### ISSUE TITLE
No code completion on obvious case

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#16257]
Godot 3.0 stable

With the following code:
```gdscript
extends Panel

func _ready():
	var f = File.new()
	f.
```

When I type the final dot, nothing comes out.
In 2.1.4 I get auto-complete straight away.

Note 1:
If I write `open("test.txt")` and attempt to Ctrl+Click on `open`, nothing happens.

Note 2:
it seems to happen in more places where completion used to work, like `onready` vars assigned with `get_node`.

**Comment by 11clock** [comment]
Noticed this with KinematicBody2D’s get_safe_margin as well.

**Comment by reduz** [comment]
functions are not completed on purpose in 3.0, only properties will
complete. Likewise only properties are shown in the help

**Comment by reduz** [comment]
let me rephrase, functions for which a property exists are not completed,
only properties

**Comment by Zylann** [comment]
@reduz why is that intentional? Those functions used to be autocompleted just fine before, now coding experience became worse :(

**Comment by reduz** [comment]
because you are not supposed to use them anymore, use the properties

**Comment by reduz** [comment]
ah, you edited it, if no property exists, regular functions should complete fine. I was refering to get_safe_margin mostly.

**Comment by vnen** [comment]
I noticed it does show the completion if you leave one (sometimes two) empty lines after the declaration, before trying to use it.

**Comment by reduz** [comment]
that definitely sounds like a bug in gdscript_editor.cpp in the complete
function checking the line

**Comment by bojidar-bg** [comment]
Can confirm, I remember it occurred "recently"

**Comment by 11clock** [comment]
@reduz How do you access the safe margin then? It has the name "collision/safe_margin" which gives me an error when I try to access it.

**Comment by Zylann** [comment]
@11clock Hmm indeed the doc is confusing about it too http://docs.godotengine.org/en/3.0/classes/class_kinematicbody2d.html#member-variables
Maybe needs a separate issue

**Comment by fuha18** [comment]
This might be a little off the topic but using a theme file generated with the "Current Editor Theme" option
also causes code completion to stop working when selected in the `Project Settings/Gui/Theme/Custom` setting.

**Comment by Scemenzo** [comment]
> also causes code completion to stop working when selected in the `Project Settings/Gui/Theme/Custom` setting.
Finally the solution I was desperately searching for these 2 days.
I'm on Godot 3.1.1 stable and I was having the "no completion" problem on a specific project on which I've changed the main theme in project settings.
The autocompletion was not working on both Windows 10 and Arch Linux.
Now everything is fine. It surely is caused by the main theme setting.

This should be re-opened.

**Comment by vnen** [comment]
@Scemenzo what you describe looks like a different bug, please open a new issue with the template filled.
